interpreter/exec/simd.ml

```diff
@@ -186,7 +188,13 @@ struct
     type t = Rep.t
     type lane = Float.t
     let unop f x = Convert.of_shape (List.map f (Convert.to_shape x))
+    let unopi f x = Convert.of_shape (List.mapi f (Convert.to_shape x))
     let binop f x y = Convert.of_shape (List.map2 f (Convert.to_shape x) (Convert.to_shape y))
+
+    let splat x = Convert.of_shape (List.init Convert.num_lanes (fun i -> x))
+    let extract_lane i s = List.nth (Convert.to_shape s) i
+    let replace_lane v i l = unopi (fun j x -> if j = i then l else x) v
```

Comment:
maybe use `x` instead of `l`, since `l` is easily confused with `I` or `1` in some fonts

Reply:
Used i/j and x/y as suggested, thanks!